test-app/build-tools/android-metadata-generator/build.gradle

```diff
@@ -0,0 +1,78 @@
+/* 
+*	Packs metadata generator in a .tgz file in ~/dist folder
+*	To build .tgz
+*			gradlew packmg
+*	To build jar
+*			gradlew jarmg
+*/
+apply plugin: "java"
+
+sourceCompatibility = 1.6
+targetCompatibility = 1.6
+
+buildscript {
+    repositories {
+        jcenter()
+    }
+
+    dependencies {
+        classpath 'com.android.tools.build:gradle:1.5.0'
+    }
+}
+
+sourceSets {
+    main {
+        java {
+            srcDir 'src/src'
+        }
+    }
+}
+
+compileJava.outputs.dir("$rootDir/dist/classes")
+
+dependencies {
+    compile files("./src/libs/bcel-5.2.jar")
+    compile files("./src/libs/dx.jar")
+}
+
+task copyNecessaryFiles {
+	doLast {
+		copy {
+			from "$rootDir/helpers"
+			into "$rootDir/dist/bin"
+		}
+		
+		copy {
+			from "$rootDir/package.json"
+			into "$rootDir/dist"
+		}	
+	}
+}
+
+jar {
+    manifest {
+        attributes("Manifest-Version": "1.0",
+                   "Main-Class": "com.telerik.metadata.Generator")
+    }
+	
+    from {
+
+        configurations.runtime.collect {
+            it.isDirectory() ? it : zipTree(it)
+        }
+
+        configurations.compile.collect {
+            it.isDirectory() ? it : zipTree(it)
+        }
+    }
+}
+
+task copyJarToBuildTools (type: Copy) {
+	from "$projectDir/build/libs/android-metadata-generator.jar"
```

Comment:
Do you think this task may be redundant, considering the runtime's build.gradle already copies the output jar to the build-tools directory - https://github.com/NativeScript/android-runtime/pull/891/files#diff-c197962302397baf3a4cc36463dce5eaR195 

Or is this for when the metadata generator project is built locally inside of Android Studio?

Reply:
It's for both. Initially the `metadatagenerator.jar` is not generated inside the `test-app/build-tools` folder, but we need it so the `app` can use it for metadata generation. So if the build is running and the jar is not there, it will be generated. The copy task you linked is moving the generated jar inside the `dist/framework...` folder so it's included in the final `tgz` package.